Getting hardware information seems very slow

When using the code and `maven` dependency from [this](https://stackoverflow.com/a/37705082/3764804) `StackOverflow` answer, the runtime of the method is extremely slow. I've seen it take 10, 15 or even 50 seconds to complete. In the meantime `OSHI` writes the following logs to the console:

```
11:42:30.781 [pool-1-thread-1] DEBUG oshi.software.os.windows.WindowsOSVersionInfoEx - Initialized OSVersionInfoEx
11:43:18.815 [pool-1-thread-1] DEBUG oshi.hardware.common.AbstractCentralProcessor - Oracle MXBean detected.
11:43:46.044 [pool-1-thread-1] DEBUG oshi.hardware.platform.windows.WindowsCentralProcessor - Initialized Processor
```

Is there a problem with my code? Is the runtime "normal" or is there some kind of performance issue? My PC is quite modern, too. In my opinion this feels too slow for "only" getting a few hardware information so this nearly makes the library obsolete for the purpose of hardware-based license key generation since users don't appreciate to wait that long. Besides this, the library is excellent of course.

Thanks a lot for your time, Mr. Widdis.

Confirmed:

> 2018-09-09 18:56:18.164 [INFO] SystemInfoTest - Checking Processor...
> 2018-09-09 18:56:18.295 [ERROR] WindowsCentralProcessor - Localized PROCESSOR: Processor
> 2018-09-09 18:56:18.297 [ERROR] WindowsCentralProcessor - Instances: [0, 1, 2, 3, _Total]
> 2018-09-09 18:56:18.300 [ERROR] WindowsCentralProcessor - I SHOULD NOT BE HERE
> 2018-09-09 18:56:51.196 [ERROR] WmiUtil - Long query: 32896 ms, SELECT [PERCENTDPCTIME, PERCENTINTERRUPTTIME, PERCENTPRIVILEGEDTIME, PERCENTPROCESSORTIME, PERCENTUSERTIME] FROM Win32_PerfRawData_PerfOS_Processor WHERE NOT Name="_Total"
> 2018-09-09 18:56:51.196 [ERROR] WindowsCentralProcessor - I SHOULD STILL NOT BE HERE

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,6 @@
 3.9.0 (in progress)
 ================
+* [#614](https://github.com/oshi/oshi/pull/614): Remove unneeded debug query and fix a WMI cast error - [@dbwiddis](https://github.com/dbwiddis).
 * Your contribution here.
 
 3.8.2 (09/07/2018)

--- a/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsCentralProcessor.java
+++ b/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsCentralProcessor.java
@@ -260,9 +260,6 @@ public class WindowsCentralProcessor extends AbstractCentralProcessor {
             interruptsQuery = new WmiQuery<>("Win32_PerfRawData_PerfOS_Processor WHERE Name=\"_Total\"",
                     InterruptsProperty.class);
         }
-        // REMOVE
-        processorTickCountQuery = new WmiQuery<>("Win32_PerfRawData_PerfOS_Processor WHERE NOT Name=\"_Total\"",
-                ProcessorTickCountProperty.class);
     }
 
     /**

--- a/oshi-core/src/main/java/oshi/software/os/windows/WindowsNetworkParams.java
+++ b/oshi-core/src/main/java/oshi/software/os/windows/WindowsNetworkParams.java
@@ -149,7 +149,7 @@ public class WindowsNetworkParams extends AbstractNetworkParams {
         int index = 0;
         int min = Integer.MAX_VALUE;
         for (int i = 0; i < vals.getResultCount(); i++) {
-            int metric = WmiUtil.getUint32(vals, NetRouteProperty.ROUTEMETRIC, i);
+            int metric = WmiUtil.getUint16(vals, NetRouteProperty.ROUTEMETRIC, i);
             if (metric < min) {
                 min = metric;
                 index = i;

